### ISSUE TITLE
bug: the pop-up does not show

### ISSUE THREAD

**Issue by technophile-04** [omerxx/tmux-sessionx#22]
### Description : 

https://github.com/omerxx/tmux-sessionx/assets/80153681/b15ad730-411d-4aab-8adc-4a6edbb9e868

For some reason, the session window does not pop open for me the screen just flickers but nothing happens : 

My tmux config : https://github.com/technophile-04/tmux.conf

My wezter config : https://github.com/technophile-04/wezterm-config

I also have fzf, bat and zoxide installed through brew

Pardon me if I missed something, also thanks for creating such an awesome plugin !!

**Comment by alexanderfast** [comment]
I have the same issue. I tried starting with `tmux -vvvv` and then launch the plugin, after seeing the selection window briefly flicker I exit and inspect the log files generated in the local folder. I can see the sessionx script being executed, but no errors. Not sure how to debug this, but ill join this thread to subscribe to eventual solutions.

**Comment by luntta** [comment]
I'm having the same issue.

**Comment by Nitestack** [comment]
I had this bug too. I switched to t-smart-tmux-manager and it didn't work too. So I looked into the issues.

I just had to kill all sessions and create a brand new session then. After that, both plugins work!

**Comment by alexanderfast** [comment]
Tried to detach and running `tmux kill-server` but that didnt work, and no matter of config reload or ssh/terminal restarts seems to affect this either.

**Comment by luntta** [comment]
> Tried to detach and running `tmux kill-server` but that didnt work, and no matter of config reload or ssh/terminal restarts seems to affect this either.

Yeah, not working for me either. Here's my setup:

OS: Ubuntu 23.04 x86_64
Kernel: 6.2.0-39-generic
Shell: zsh 5.9
Resolution: 2560x1440, 2560x1440
DE: GNOME 44.3
Terminal: tmux 3.3a

**Comment by Nitestack** [comment]
So you guys killed each session (so that no client is running) and it didnt work? Strange...

**Comment by johnmosv** [comment]
I have the same issue on mac with kitty. Tried detaching and killing the server without any effect.

**Comment by Chris44442** [comment]
Same issue here. Right now I'm pretty sure it has something to do with fzf. On my one machine I could make it work by installing the binary from their github and uninstalling the apt version of fzf and doing a bunch of tmux kill-server. But on my other machine this didn't work for some reason. Very hard to pinpoint exactly.

**Comment by alexanderfast** [comment]
Just checked and this is my version:

```
❯ fzf --version
0.38.0 (debian)
```

I found an alternate session manager for tmux so i wont spend super much more time on this, but ill stay subscribed to the issue and ill check again when/if it's fixed. :)

**Comment by the-spanish-guy** [comment]
Any updates? I have the same problem :/

**Comment by TechDufus** [comment]
What version of `fzf` is everyone using here? `fzf --version`
You may want to consider updating, since the `fzf` available in `apt` only installs v0.39 or so, and after I updated from the `fzf` github release this worked for me.

https://github.com/omerxx/tmux-sessionx/issues/9
https://github.com/omerxx/tmux-sessionx/issues/12

**Comment by the-spanish-guy** [comment]
> What version of `fzf` is everyone using here? `fzf --version` You may want to consider updating, since the `fzf` available in `apt` only installs v0.39 or so, and after I updated from the `fzf` github release this worked for me.
> 
> #9 #12

is v0.43.0
![image](https://github.com/omerxx/tmux-sessionx/assets/32144508/473d7ca0-3598-4756-9b6d-d755c9e3d957)

**Comment by omerxx** [comment]
Hi friends!
I'll try to figure out the issue you're facing, but in the mean time try using this option, fzf is missing a few features in older versions that I'm using now (by older I mean two years at least).
```
# If you're running fzf lower than 0.35.0 there are a few missing features
# Upgrade, or use this setting for support
set -g @sessionx-legacy-fzf-support 'on'
```

If you guys confirm this does the trick I think I'll either remove the functionality as it's not all that important, or better yet, make it ope-in rather than out.

Thanks!

**Comment by omerxx** [comment]
> Just checked and this is my version:
> 
> ```
> ❯ fzf --version
> 0.38.0 (debian)
> ```
> 
> I found an alternate session manager for tmux so i wont spend super much more time on this, but ill stay subscribed to the issue and ill check again when/if it's fixed. :)

Can you try to either upgrade your fzf or use `set -g @sessionx-legacy-fzf-support 'on'`?

**Comment by omerxx** [comment]
> Any updates? I have the same problem :/

Can you please share your relevant config? 
Seems like you're running a relatively up-to-date fzf so I'm curious why your setup fails

**Comment by omerxx** [comment]
> ### Description :
>  Screen.Recording.2023-12-30.at.3.28.26.PM.mov 
> For some reason, the session window does not pop open for me the screen just flickers but nothing happens :
> 
> My tmux config : https://github.com/technophile-04/tmux.conf
> 
> My wezter config : https://github.com/technophile-04/wezterm-config
> 
> I also have fzf, bat and zoxide installed through brew
> 
> Pardon me if I missed something, also thanks for creating such an awesome plugin !!

I used your config with no issues, however the plugin is not there (assuming this is why I see an unstaged git change in your video above) so can't replicate everything.
Please try these - 

- [ ] Make sure your fzf version is recent (at least > `0.35`)
- [ ] Try setting a different key with `set -g @sessionx-bind '<mykey>'`
- [ ] Remove all other possible conflicts (like the t-session-manager) is you can. (`<prefix>+alt+u`) should cleanup after you've removed them
- [ ] Try resetting the server after the above

**Comment by omerxx** [comment]
> Same issue here. Right now I'm pretty sure it has something to do with fzf. On my one machine I could make it work by installing the v0.45.0 binary from their github and uninstalling the apt version of fzf and doing a bunch of tmux kill-server. But on my other machine the tmux crashes when i open sessionx with fzf 0.45. When I comment out the `RESULT=$(echo -e "${INPUT// /}" | fzf-tmux "${args[@]}")` line, it doesn't crash (but doesn't work either obviously).
> 
> Edit: One thing to note is: "zoxide only supports fzf v0.33.0 and above". Apt version is v0.29.0-1. The Apt version is where the flickering happens.

Thanks for clarifying!
So do you think there's an issue with the apt version across the board?
If that's really the case my idea with ignoring relatively new fzf features won't do...
I wonder whats the best approach here... maybe a preflight check for version or something like that 🤔

**Comment by johnmosv** [comment]
I can verify updating fzf did it for me. thanks alot!

**Comment by b1tray3r** [comment]
Hi folks!

Found the YouTube video and wanted to install this session manager to my tmux setup... and ended up here in this issue because the popup window didn't open.

After someone in the comments mentioned a log file, I went into the plugins folder to search for such...

I ended up in the scripts' subfolder at this line:

https://github.com/omerxx/tmux-sessionx/blob/main/scripts/sessionx.sh#L16C52-L16C73

Please make sure to enable the pop-up window because it is disabled by default:

```config
set -g @sessionx-window-mode 'on'
```

Kind regards

Alex

**Comment by the-spanish-guy** [comment]
> Hi folks!
> 
> Found the YouTube video and wanted to install this session manager to my tmux setup... and ended up here in this issue because the popup window didn't open.
> 
> After someone in the comments mentioned a log file, I went into the plugins folder to search for such...
> 
> I ended up in the scripts' subfolder at this line:
> 
> https://github.com/omerxx/tmux-sessionx/blob/main/scripts/sessionx.sh#L16C52-L16C73
> 
> Please make sure to enable the pop-up window because it is disabled by default:
> 
> ```
> set -g @sessionx-window-mode 'on'
> ```
> 
> Kind regards
> 
> Alex
> 
> P.S.: This may be added in the documentation, because I ended up installing all the requirements... The YouTube video says "Just add this line..." but this ain't the truth. :D

I set this setting but it still didn't work :/

**Comment by omurilo** [comment]
Hi guys, any update here? I'm in same problem within Ubuntu and MacOS.

**Comment by the-spanish-guy** [comment]
I did another test keeping the configuration on Windows (wsl2 with arch) and it worked, I'll try again on Ubuntu later and post updates here

**Comment by TechDufus** [comment]
> I did another test keeping the configuration on Windows (wsl2 with arch) and it worked, I'll try again on Ubuntu later and post updates here

came here to call out this [arch btw] btw.

**Comment by the-spanish-guy** [comment]
I uninstalled and reinstalled tmux, leaving only tpm and tmux-sessionx as plugins and it didn't work

**Comment by omerxx** [comment]
Hi everyone, I'd really want to get to the bottom of this.
Happy if anyone here wants to jump on a call / try and share their setup.

I'm running latest Tmux + FZF + Zoxide.
I use Wezterm and the only time I was able to reproduce it was when the plugin script had a bug / fzf was not updated.

If anyone's interested please comment here or if you use Telegram let's chat directly: https://t.me/omerxx

**Comment by the-spanish-guy** [comment]
I downloaded and installed Wezterm and it worked, I also tested it with Ubuntu Terminal and it worked so I decided to test it one last time with Kitty and it worked too

![image](https://github.com/omerxx/tmux-sessionx/assets/32144508/9f79e4d3-5485-43f3-887d-a9fc38d29d6f)

this is my config
```shell
unbind-key o

# List of plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'omerxx/tmux-sessionx'
set -g @plugin 'catppuccin/tmux'

# I recommend using `o` if not already in use, for least key strokes when launching
set -g @sessionx-bind 'o'
set -g @sessionx-window-mode 'on'


# Other examples:
# set -g @plugin 'github_username/plugin_name'
# set -g @plugin 'github_username/plugin_name#branch'
# set -g @plugin 'git@github.com:user/plugin'
# set -g @plugin 'git@bitbucket.com:user/plugin'

# set -g @catppuccin_status_modules_right "... battery ..."
set -g @catppuccin_window_left_separator ""
set -g @catppuccin_window_right_separator " "
set -g @catppuccin_window_middle_separator " █"
set -g @catppuccin_window_number_position "right"

set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W"

set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#W"

set -g @catppuccin_status_modules_right "session battery"
set -g @catppuccin_status_left_separator  " "
set -g @catppuccin_status_right_separator ""
set -g @catppuccin_status_right_separator_inverse "no"
set -g @catppuccin_status_fill "icon"
set -g @catppuccin_status_connect_separator "no"

set -g @catppuccin_directory_text "#{pane_current_path}"

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '~/.tmux/plugins/tpm/tpm'

```
I was confused as to why it worked now but it is working

**Comment by b1tray3r** [comment]
Well I am running this setup, where it is running now:

❯ tmux -V
tmux 3.3a
❯ zoxide -V
zoxide v0.4.3-unknown
❯ fzf --version
0.45.0 (202401011)

~/.config/tmux/tmux.conf
```Conf
# [...] unrelated
set -g @sessionx-bind 'O'
set -g @sessionx-window-mode 'on'
# [...] unrelated
```


System information:

OS: Ubuntu 23.04 lunar
Kernel: x86_64 Linux 6.2.0-39-generic
Shell: zsh 5.9

**Comment by omurilo** [comment]
Damm, now Im stucked when press preffix + o, need open new terminal tab and close the actual.

**Comment by the-spanish-guy** [comment]
> Damm, agora estou preso quando pressione preffix + o, precisa abrir nova guia terminal e fechar o real.

It happened to me too, did you do `unbind-key o` before `set -g @sessionx-bind 'o'` ?

**Comment by omerxx** [comment]
Hi people, anyone still having issues and would like to share workflow / setup?
Reply here / DM at https://t.me/omerxx

**Comment by enthusiva** [comment]
In my case, one of the prerequisites "fzf-tmux" script of this plugin was not installed in my system. once I added it, I could see the pop-up.

https://github.com/junegunn/fzf/blob/master/bin/fzf-tmux

**Comment by touchedthecode** [comment]
I came here because of the same problem, on my Mac it worked fine, but on **Windows 10 WSL2 Ubuntu** it didn't work. I tried everything but nothing worked, until I randomly fixed it by switching from zsh to bash and back to zsh. 

Maybe it was some kind of caching problem, although I restarted zsh multiple times and that didn't help. Weird. 

Maybe this can help someone. :)

**Comment by moritz157** [comment]
I have the same problem on PopOS with zsh.
- fzf version: 0.45.0
- tmux version: 3.2a
- fzf-tmux is installed
- zoxide version: 0.9.3
- bat version: 0.19.0

I used exactly the same configuration @the-spanish-guy used, restarted tmux and even tried rebooting, but it didn't work.
Tmux completely freezes, when I press the keybinding. However, I can close the terminal, open a new one, reattach to the previously frozen session and it unfreezes again.
I also tried rebinding to a different key, but that didn't help either.

**Comment by moritz157** [comment]
Update: I got it to work 🚀 
Looks like there is a problem with tmux version `3.2a` on certain Ubuntu releases as hinted at in [this issue](https://github.com/joshmedeski/t-smart-tmux-session-manager/issues/104#issuecomment-1889778990) on the t-smart-tmux-session-manager project.

However, since `3.2a` is the newest version installable through apt, I had to build tmux from source.
I am now on tmux version `3.3a` and everything works fine.

**Comment by conghuynhho** [comment]
Update all of these make it works for me:
Tmux + FZF + Zoxide.

**Comment by MoS2DL** [comment]
I experienced the same issue here, no pop up showing. But managed to fix it.

I'm on Pop!_os and had the apt version of tmux installed, that is version 3.2a.

Seeing the last comments here I tried to uninstall tmux 3.2a and compiled version 3.4 from source instead. Now the popup works. So, it seems that tmux 3.2a has some kind of conflict at least.

**Comment by Zananok** [comment]
Same issue.
Freezes on prefix + O. Restore access by closing Terminal/tab and reopening, as in previous reports.

```
# [...] unrelated
set -g @sessionx-bind 'O'
set -g @sessionx-window-mode 'on'
# [...] unrelated
```

Tried with standard latest debian packages, then manually updated/built everything with no success:
```
❯ zoxide -V
zoxide 0.9.4
❯ tmux -V
tmux next-3.5
❯ fzf --version
0.46.1 (3c0a630)
❯ fzf-tmux --version
fzf-tmux (with fzf 0.46.1 (3c0a630))
❯ bat --version
bat 0.24.0 (fc95468)
❯ zsh --version
zsh 5.9 (x86_64-pc-linux-gnu)
❯ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
```
Windows 11 64x - WSL v1 - Debian 11 - Windows Terminal 1.19.10573.0

**Comment by ProgmRuanSilva** [comment]
Hey @Zananok

Did you test the default or another keys?
Or change the @sessionx-bind to be another key?

Tanks to your contribution

**Comment by Zananok** [comment]
Hey, @ProgmRuanSilva tried with both o and O, and now, for a sanity check, with both u and r. Additionally, I removed any "@sessionx-bind", restarted the tmux server and attempted using the default o, however it seemed to just do a default tmux command of switching between panes. By default, I start tmux with -u, but I also killed the server and started it with no flags and attempted the same experiment as aforementioned.

All attempts resulted in the same outcome: gets frozen, with the exception of the unset "@sessionx-bind", but none resulted in success.

Thank you for your time

**Comment by ProgmRuanSilva** [comment]
understood,

let's try these options:

1. Check if the tmux and fzf  versions, and update it to the newest versions
2. Might is the bind keys which is unreally in uses
    - unset all key using: `unbind -a`
    - see if it works
    - if not set the rebind keys.

**Comment by Andreasgdp** [comment]
For me tmux-sessionx would not appear at all if tmux was initially launched (i.e. no active sessions) as a command when opening a terminal e.g. `terminator -e "tmux"` or `kitty sh -c "tmux"`...

For me i solved it with adding the following to my `.zshrc` file.
`if [ "$TMUX" = "" ]; then tmux; fi`

It seemed to also work with just starting tmux from a blank terminal using `tmux` if you don't want to have it launch every time.

System: Ubuntu 22.04
I also built tmux from source (v. tmux next-3.5), so that can also have something to do with it. But just building from source did not fix. Only after also doing the above.

**Comment by ewilazarus** [comment]
Hey everybody!

In case you're autostarting tmux in your shell, make sure that you source the `.fzf.zsh`/`.fzf.bash` before you do so.

This is what was breaking my configuration! I hope this helps. :wave:

**Comment by omerxx** [comment]
Hello good people of this thread, just wondering whether anyone still experiencing issues with this.
If not I'll give this issue a few more days before closing it.

**Comment by Zananok** [comment]
The issue is how the fzf args were handled. I did a quick fix locally, most of which was to wrap the args in quotes. But decided to go on to a custom solution.

I checked your sessionx.sh and now it seems that they were wrapped in ", so I decided to try it once more from main and it works for me.

**Comment by omerxx** [comment]
> The issue is how the fzf args were handled. I did a quick fix locally, most of which was to wrap the args in quotes. But decided to go on to a custom solution.
> 
> 
> 
> I checked your sessionx.sh and now it seems that they were wrapped in ", so I decided to try it once more from main and it works for me.

Thank you!

**Comment by omerxx** [comment]
Please reopen if this is becomes relevant again

**Comment by hesperaux** [comment]
Hi all,
I'm seeing this issue. If I watch very carefully and attempt to open it repeatedly, I can see a brief glimpse inside the popup with an error message saying something like "scrollbar display width should be 1" or something to that effect. It's so fast I can't capture it.

I'm running fzf 0.38.0 (debian), with tmux 3.3a. I'm on debian linux (bookworm 12.5) with alacritty 0.11.0.

Any suggestions?

Thanks!
-David

**Comment by omerxx** [comment]
> Hi all, I'm seeing this issue. If I watch very carefully and attempt to open it repeatedly, I can see a brief glimpse inside the popup with an error message saying something like "scrollbar display width should be 1" or something to that effect. It's so fast I can't capture it.
> 
> I'm running fzf 0.38.0 (debian), with tmux 3.3a. I'm on debian linux (bookworm 12.5) with alacritty 0.11.0.
> 
> Any suggestions?
> 
> Thanks! -David

Hey @hesperaux, the issue stems from the terminal's inability to draw the scrollbar - https://github.com/omerxx/tmux-sessionx/blob/56abffd0155805f8165fb5af19c50d04c4a49069/scripts/sessionx.sh#L209C16-L209C18

I don't think this is an Alacritty issue. Do you have Nerdfonts installed?
https://www.nerdfonts.com/

**Comment by hesperaux** [comment]
@omerxx thanks for the info. I do have nerd fonts installed (0xproto nerd font).

Are you saying that the terminal should support a scroll bar, or that it doesn't matter if it does?

I started to drive into the scripts earlier but I kinda suck at bash and I'm not familiar with tmux plugin dev either. I was going to try installing fzf from git to try that but I ran out of time for the night.

**Comment by omerxx** [comment]
@hesperaux sorry, should have added another comment, but I edited mine.
Nope, the scollbar isn't part of nerdfonts, it's a simple unicode. 
For whatever reason the scrollbar string fzf is getting on your env is empty, which is why I'm suggesting
1. Make sure FZF is updated (will likely help)
2. Make sure the plugin is updated (less likely the reason)

**Comment by hesperaux** [comment]
@omerxx Thank you for taking the time to help.
It's working great now. I realized that I had fzf installed via git but it was not the one being used. It was obvious but I overlooked it. I had installed fzf from git and it was in my home directory bin folder, but because it had been installed by apt into /usr/bin, that path was taking precedance. The version from git is much newer (0.46.1).
That took care of it. This looks like a nice plugin. Happy I can finally use it. :)

**Comment by bnZuhair** [comment]
> ```shell
> unbind-key o
> ```

does it have to be installed to a specific location?

**Comment by Smalto22** [comment]
I may be missing something or this issue might have resurfaced... So, please forgive my ignorance.
I started experiencing this yesterday, after I reinstalled tmux and some other packages.
Here are the versions of the packages I have installed:

- tmux: stable 3.5a (bottled), HEAD
- fzf: 0.57.0 (brew)
- zoxide: 0.9.6
- bat: 0.24.0

I am not quite familiar with the bash language but here is the output I get when I manually run the `sessionz.sh` script with `bash -x  ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh`:

```
+++ dirname ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh
++ cd ./tmux/plugins/tmux-sessionx/scripts
++ pwd
+ CURRENT_DIR=/Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts
++ tmux display-message -p '#S'
+ CURRENT=dev
+ Z_MODE=off
+ source /Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts/tmuxinator.sh
+ source /Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts/fzf-marks.sh
+ run_plugin
++ tmux show-option -gqv @sessionx-_built-args
+ eval declare -a 'args='\''([0]="--bind"' '[1]="ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' -t '{1})"' '[2]="--bind"' '[3]="ctrl-x:reload(find' '~/Dotfiles' -mindepth 1 -maxdepth 1 -type d -o -type 'l)+change-preview(ls' '{})"' '[4]="--bind"' '[5]="ctrl-w:reload(tmux' list-windows -a -F ''\''\'\'''\''#{session_name}:#{window_name}'\''\'\'''\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' -w '{1})"' '[6]="--bind"' '[7]="ctrl-e:reload(find' /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type 'l)+change-preview(ls' '{})"' '[8]="--bind"' '[9]="ctrl-y:reload(zoxide' query '-l)+change-preview(ls' '{})"' '[10]="--bind"' '[11]="alt-bspace:execute-silent(tmux' kill-session -t '{})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)"' '[12]="--bind"' '[13]="bspace:backward-delete-char"' '[14]="--bind"' '[15]="esc:abort"' '[16]="--bind"' '[17]="ctrl-n:up"' '[18]="--bind"' '[19]="ctrl-p:down"' '[20]="--bind"' '[21]="enter:replace-query+print-query"' '[22]="--bind"' '[23]="ctrl-u:preview-half-page-up"' '[24]="--bind"' '[25]="ctrl-d:preview-half-page-down"' '[26]="--bind"' '[27]="ctrl-r:execute(bash' -c ''\''\'\'''\''' printf '>&2' '\"New' name: '\";read' 'name;' tmux rename-session -t '{1}' '\"\${name}\";' ''\''\'\'''\'')+reload(bash' -c ''\''\'\'''\''' tmux list-sessions '|' sed -E '\"s/:.*\$//\";' ''\''\'\'''\'')"' '[28]="--bind"' '[29]="?:toggle-preview"' '[30]="--bind"' '[31]="change:first"' '[32]="--exit-0"' '[33]="--header=enter=󰿄' $'\202' $'alt-bspace=\202' $'\202' $'ctrl-r=\221\225' $'\202' $'ctrl-x=\203\226' $'\202' $'ctrl-w=\213\222' $'\202' $'ctrl-e=\207\230' $'\202' $'ctrl-b=\214\215' $'\202' $'ctrl-t=\220\206' $'\202' $'ctrl-u=\204' $'\202' $'ctrl-d=\204' / 'ctrl-y="' '[34]="--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' '{}"' '[35]="--preview-window=top,75%,,"' '[36]="--layout=default"' '[37]="--pointer=▶"' '[38]="-p"' '[39]="75%,85%"' '[40]="--prompt"' '[41]="' '"' '[42]="--print-query"' '[43]="--tac"' '[44]="--scrollbar"' '[45]="▌▐"' '[46]="--border-label"' '[47]="Current' session: '\"\"' '"' '[48]="--bind"' '[49]="focus:transform-preview-label:echo' '[' '{}' ']")'\'''
++ declare -a 'args=([0]="--bind" [1]="ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -t {1})" [2]="--bind" [3]="ctrl-x:reload(find ~/Dotfiles -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})" [4]="--bind" [5]="ctrl-w:reload(tmux list-windows -a -F '\''#{session_name}:#{window_name}'\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -w {1})" [6]="--bind" [7]="ctrl-e:reload(find /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})" [8]="--bind" [9]="ctrl-y:reload(zoxide query -l)+change-preview(ls {})" [10]="--bind" [11]="alt-bspace:execute-silent(tmux kill-session -t {})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)" [12]="--bind" [13]="bspace:backward-delete-char" [14]="--bind" [15]="esc:abort" [16]="--bind" [17]="ctrl-n:up" [18]="--bind" [19]="ctrl-p:down" [20]="--bind" [21]="enter:replace-query+print-query" [22]="--bind" [23]="ctrl-u:preview-half-page-up" [24]="--bind" [25]="ctrl-d:preview-half-page-down" [26]="--bind" [27]="ctrl-r:execute(bash -c '\'' printf >&2 \"New name: \";read name; tmux rename-session -t {1} \"\${name}\"; '\'')+reload(bash -c '\'' tmux list-sessions | sed -E \"s/:.*\$//\"; '\'')" [28]="--bind" [29]="?:toggle-preview" [30]="--bind" [31]="change:first" [32]="--exit-0" [33]="--header=enter=󰿄  alt-bspace=󱂧  ctrl-r=󰑕  ctrl-x=󱃖  ctrl-w=  ctrl-e=󰇘  ctrl-b=󰌍  ctrl-t=󰐆  ctrl-u=  ctrl-d= / ctrl-y=" [34]="--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {}" [35]="--preview-window=top,75%,," [36]="--layout=default" [37]="--pointer=▶" [38]="-p" [39]="75%,85%" [40]="--prompt" [41]=" " [42]="--print-query" [43]="--tac" [44]="--scrollbar" [45]="▌▐" [46]="--border-label" [47]="Current session: \"\" " [48]="--bind" [49]="focus:transform-preview-label:echo [ {} ]")'
++ tmux show-option -gqv @sessionx-_built-extra-options
+ eval declare -a 'extra_options='\''([0]="false")'\'''
++ declare -a 'extra_options=([0]="false")'
+ handle_input
++ input
++ default_window_mode=false
++ [[ false == \o\n ]]
++ filter_current_session=false
++ [[ false == \t\r\u\e ]]
++ get_sorted_sessions
+++ tmux display-message -p '#{client_last_session}'
++ last_session=
+++ tmux list-sessions
+++ sed -E 's/:.*$//'
+++ grep -v '^$'
++ sessions='alpha
dev'
./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
dev: syntax error in expression (error token is "dev")
++ echo dev
+ INPUT=dev
++ additional_input
+++ get_sorted_sessions
++++ tmux display-message -p '#{client_last_session}'
+++ last_session=
++++ tmux list-sessions
++++ sed -E 's/:.*$//'
++++ grep -v '^$'
+++ sessions='alpha
dev'
./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
dev: syntax error in expression (error token is "dev")
++ sessions=
++ custom_paths=false
++ custom_path_subdirectories=false
++ [[ -z false ]]
+++ echo false
+++ sed -E 's/ *, */,/g'
+++ sed -E 's/^ *//'
+++ sed -E 's/ *$//'
+++ sed -E 's/ /✗/g'
++ clean_paths=false
++ [[ false == \t\r\u\e ]]
++ paths=false
++ export -f add_path
++ printf '%s\n' false
++ xargs -n 1 -P 0 bash -c 'add_path "$@"' _
+ ADDITIONAL_INPUT=false
+ [[ -n false ]]
++ additional_input
+++ get_sorted_sessions
++++ tmux display-message -p '#{client_last_session}'
+++ last_session=
++++ tmux list-sessions
++++ sed -E 's/:.*$//'
++++ grep -v '^$'
+++ sessions='alpha
dev'
./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
dev: syntax error in expression (error token is "dev")
++ sessions=
++ custom_paths=false
++ custom_path_subdirectories=false
++ [[ -z false ]]
+++ echo false
+++ sed -E 's/ *, */,/g'
+++ sed -E 's/^ *//'
+++ sed -E 's/ *$//'
+++ sed -E 's/ /✗/g'
++ clean_paths=false
++ [[ false == \t\r\u\e ]]
++ paths=false
++ export -f add_path
++ printf '%s\n' false
++ xargs -n 1 -P 0 bash -c 'add_path "$@"' _
+ INPUT='false\ndev'
+ bind_back=false
+ BACK='false:reload(echo -e "false\ndev")+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {1})'
+ args+=(--bind "$BACK")
+ [[ '' == \o\n ]]
++ echo -e 'false\ndev'
++ sed -E 's/✗/ /g'
++ tail -n1
++ fzf-tmux --bind 'ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -t {1})' --bind 'ctrl-x:reload(find ~/Dotfiles -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})' --bind 'ctrl-w:reload(tmux list-windows -a -F '\''#{session_name}:#{window_name}'\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -w {1})' --bind 'ctrl-e:reload(find /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})' --bind 'ctrl-y:reload(zoxide query -l)+change-preview(ls {})' --bind 'alt-bspace:execute-silent(tmux kill-session -t {})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)' --bind bspace:backward-delete-char --bind esc:abort --bind ctrl-n:up --bind ctrl-p:down --bind enter:replace-query+print-query --bind ctrl-u:preview-half-page-up --bind ctrl-d:preview-half-page-down --bind 'ctrl-r:execute(bash -c '\'' printf >&2 "New name: ";read name; tmux rename-session -t {1} "${name}"; '\'')+reload(bash -c '\'' tmux list-sessions | sed -E "s/:.*$//"; '\'')' --bind '?:toggle-preview' --bind change:first --exit-0 '--header=enter=󰿄  alt-bspace=󱂧  ctrl-r=󰑕  ctrl-x=󱃖  ctrl-w=  ctrl-e=󰇘  ctrl-b=󰌍  ctrl-t=󰐆  ctrl-u=  ctrl-d= / ctrl-y=' '--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {}' --preview-window=top,75%,, --layout=default $'--pointer=\226' -p 75%,85% --prompt ' ' --print-query --tac --scrollbar $'\226\214\226\220' --border-label 'Current session: "" ' --bind 'focus:transform-preview-label:echo [ {} ]' --bind 'false:reload(echo -e "false\ndev")+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {1})'
+ RESULT=
+ handle_output ''
+ '[' -d '' ']'
+ is_fzf-marks_mark ''
++ echo ''
++ grep -q ' : '
+ return 1
+ echo ''
+ grep :
++ echo ''
++ tr -d '\n'
+ target=
+ [[ -z '' ]]
+ exit 0
```

**Comment by Smalto22** [comment]
Never mind, this was fixed by updating my bash version as described here: https://github.com/omerxx/tmux-sessionx/issues/163#issuecomment-2564827113

**Comment by rjorgenson** [comment]
I don't suppose anyone else has run into this issue? I am on the latest bash in homebrew (`5.2.37(1)-release`) and I am having this exact issue so it doesn't seem to be related to the bash version. I have two running tmux sessions named `1` and `2`. If I have only a single session the script just flashes really quickly and goes away with no output and a `0` return code.

```
#: bash --version
GNU bash, version 5.2.37(1)-release (aarch64-apple-darwin24.2.0)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
#: bash .tmux/plugins/tmux-sessionx/scripts/sessionx.sh
.tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
2: syntax error in expression (error token is "2")
.tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
2: syntax error in expression (error token is "2")
.tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
2: syntax error in expression (error token is "2")
#:
```

> I may be missing something or this issue might have resurfaced... So, please forgive my ignorance. I started experiencing this yesterday, after I reinstalled tmux and some other packages. Here are the versions of the packages I have installed:
> 
> * tmux: stable 3.5a (bottled), HEAD
> * fzf: 0.57.0 (brew)
> * zoxide: 0.9.6
> * bat: 0.24.0
> 
> I am not quite familiar with the bash language but here is the output I get when I manually run the `sessionz.sh` script with `bash -x ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh`:
> 
> ```
> +++ dirname ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh
> ++ cd ./tmux/plugins/tmux-sessionx/scripts
> ++ pwd
> + CURRENT_DIR=/Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts
> ++ tmux display-message -p '#S'
> + CURRENT=dev
> + Z_MODE=off
> + source /Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts/tmuxinator.sh
> + source /Users/my-user/Dotfiles/tmux/plugins/tmux-sessionx/scripts/fzf-marks.sh
> + run_plugin
> ++ tmux show-option -gqv @sessionx-_built-args
> + eval declare -a 'args='\''([0]="--bind"' '[1]="ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' -t '{1})"' '[2]="--bind"' '[3]="ctrl-x:reload(find' '~/Dotfiles' -mindepth 1 -maxdepth 1 -type d -o -type 'l)+change-preview(ls' '{})"' '[4]="--bind"' '[5]="ctrl-w:reload(tmux' list-windows -a -F ''\''\'\'''\''#{session_name}:#{window_name}'\''\'\'''\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' -w '{1})"' '[6]="--bind"' '[7]="ctrl-e:reload(find' /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type 'l)+change-preview(ls' '{})"' '[8]="--bind"' '[9]="ctrl-y:reload(zoxide' query '-l)+change-preview(ls' '{})"' '[10]="--bind"' '[11]="alt-bspace:execute-silent(tmux' kill-session -t '{})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)"' '[12]="--bind"' '[13]="bspace:backward-delete-char"' '[14]="--bind"' '[15]="esc:abort"' '[16]="--bind"' '[17]="ctrl-n:up"' '[18]="--bind"' '[19]="ctrl-p:down"' '[20]="--bind"' '[21]="enter:replace-query+print-query"' '[22]="--bind"' '[23]="ctrl-u:preview-half-page-up"' '[24]="--bind"' '[25]="ctrl-d:preview-half-page-down"' '[26]="--bind"' '[27]="ctrl-r:execute(bash' -c ''\''\'\'''\''' printf '>&2' '\"New' name: '\";read' 'name;' tmux rename-session -t '{1}' '\"\${name}\";' ''\''\'\'''\'')+reload(bash' -c ''\''\'\'''\''' tmux list-sessions '|' sed -E '\"s/:.*\$//\";' ''\''\'\'''\'')"' '[28]="--bind"' '[29]="?:toggle-preview"' '[30]="--bind"' '[31]="change:first"' '[32]="--exit-0"' '[33]="--header=enter=󰿄' $'\202' $'alt-bspace=\202' $'\202' $'ctrl-r=\221\225' $'\202' $'ctrl-x=\203\226' $'\202' $'ctrl-w=\213\222' $'\202' $'ctrl-e=\207\230' $'\202' $'ctrl-b=\214\215' $'\202' $'ctrl-t=\220\206' $'\202' $'ctrl-u=\204' $'\202' $'ctrl-d=\204' / 'ctrl-y="' '[34]="--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh' '{}"' '[35]="--preview-window=top,75%,,"' '[36]="--layout=default"' '[37]="--pointer=▶"' '[38]="-p"' '[39]="75%,85%"' '[40]="--prompt"' '[41]="' '"' '[42]="--print-query"' '[43]="--tac"' '[44]="--scrollbar"' '[45]="▌▐"' '[46]="--border-label"' '[47]="Current' session: '\"\"' '"' '[48]="--bind"' '[49]="focus:transform-preview-label:echo' '[' '{}' ']")'\'''
> ++ declare -a 'args=([0]="--bind" [1]="ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -t {1})" [2]="--bind" [3]="ctrl-x:reload(find ~/Dotfiles -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})" [4]="--bind" [5]="ctrl-w:reload(tmux list-windows -a -F '\''#{session_name}:#{window_name}'\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -w {1})" [6]="--bind" [7]="ctrl-e:reload(find /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})" [8]="--bind" [9]="ctrl-y:reload(zoxide query -l)+change-preview(ls {})" [10]="--bind" [11]="alt-bspace:execute-silent(tmux kill-session -t {})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)" [12]="--bind" [13]="bspace:backward-delete-char" [14]="--bind" [15]="esc:abort" [16]="--bind" [17]="ctrl-n:up" [18]="--bind" [19]="ctrl-p:down" [20]="--bind" [21]="enter:replace-query+print-query" [22]="--bind" [23]="ctrl-u:preview-half-page-up" [24]="--bind" [25]="ctrl-d:preview-half-page-down" [26]="--bind" [27]="ctrl-r:execute(bash -c '\'' printf >&2 \"New name: \";read name; tmux rename-session -t {1} \"\${name}\"; '\'')+reload(bash -c '\'' tmux list-sessions | sed -E \"s/:.*\$//\"; '\'')" [28]="--bind" [29]="?:toggle-preview" [30]="--bind" [31]="change:first" [32]="--exit-0" [33]="--header=enter=󰿄  alt-bspace=󱂧  ctrl-r=󰑕  ctrl-x=󱃖  ctrl-w=  ctrl-e=󰇘  ctrl-b=󰌍  ctrl-t=󰐆  ctrl-u=  ctrl-d= / ctrl-y=" [34]="--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {}" [35]="--preview-window=top,75%,," [36]="--layout=default" [37]="--pointer=▶" [38]="-p" [39]="75%,85%" [40]="--prompt" [41]=" " [42]="--print-query" [43]="--tac" [44]="--scrollbar" [45]="▌▐" [46]="--border-label" [47]="Current session: \"\" " [48]="--bind" [49]="focus:transform-preview-label:echo [ {} ]")'
> ++ tmux show-option -gqv @sessionx-_built-extra-options
> + eval declare -a 'extra_options='\''([0]="false")'\'''
> ++ declare -a 'extra_options=([0]="false")'
> + handle_input
> ++ input
> ++ default_window_mode=false
> ++ [[ false == \o\n ]]
> ++ filter_current_session=false
> ++ [[ false == \t\r\u\e ]]
> ++ get_sorted_sessions
> +++ tmux display-message -p '#{client_last_session}'
> ++ last_session=
> +++ tmux list-sessions
> +++ sed -E 's/:.*$//'
> +++ grep -v '^$'
> ++ sessions='alpha
> dev'
> ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
> dev: syntax error in expression (error token is "dev")
> ++ echo dev
> + INPUT=dev
> ++ additional_input
> +++ get_sorted_sessions
> ++++ tmux display-message -p '#{client_last_session}'
> +++ last_session=
> ++++ tmux list-sessions
> ++++ sed -E 's/:.*$//'
> ++++ grep -v '^$'
> +++ sessions='alpha
> dev'
> ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
> dev: syntax error in expression (error token is "dev")
> ++ sessions=
> ++ custom_paths=false
> ++ custom_path_subdirectories=false
> ++ [[ -z false ]]
> +++ echo false
> +++ sed -E 's/ *, */,/g'
> +++ sed -E 's/^ *//'
> +++ sed -E 's/ *$//'
> +++ sed -E 's/ /✗/g'
> ++ clean_paths=false
> ++ [[ false == \t\r\u\e ]]
> ++ paths=false
> ++ export -f add_path
> ++ printf '%s\n' false
> ++ xargs -n 1 -P 0 bash -c 'add_path "$@"' _
> + ADDITIONAL_INPUT=false
> + [[ -n false ]]
> ++ additional_input
> +++ get_sorted_sessions
> ++++ tmux display-message -p '#{client_last_session}'
> +++ last_session=
> ++++ tmux list-sessions
> ++++ sed -E 's/:.*$//'
> ++++ grep -v '^$'
> +++ sessions='alpha
> dev'
> ./tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: alpha
> dev: syntax error in expression (error token is "dev")
> ++ sessions=
> ++ custom_paths=false
> ++ custom_path_subdirectories=false
> ++ [[ -z false ]]
> +++ echo false
> +++ sed -E 's/ *, */,/g'
> +++ sed -E 's/^ *//'
> +++ sed -E 's/ *$//'
> +++ sed -E 's/ /✗/g'
> ++ clean_paths=false
> ++ [[ false == \t\r\u\e ]]
> ++ paths=false
> ++ export -f add_path
> ++ printf '%s\n' false
> ++ xargs -n 1 -P 0 bash -c 'add_path "$@"' _
> + INPUT='false\ndev'
> + bind_back=false
> + BACK='false:reload(echo -e "false\ndev")+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {1})'
> + args+=(--bind "$BACK")
> + [[ '' == \o\n ]]
> ++ echo -e 'false\ndev'
> ++ sed -E 's/✗/ /g'
> ++ tail -n1
> ++ fzf-tmux --bind 'ctrl-t:change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -t {1})' --bind 'ctrl-x:reload(find ~/Dotfiles -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})' --bind 'ctrl-w:reload(tmux list-windows -a -F '\''#{session_name}:#{window_name}'\'')+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh -w {1})' --bind 'ctrl-e:reload(find /Users/my-user -mindepth 1 -maxdepth 1 -type d -o -type l)+change-preview(ls {})' --bind 'ctrl-y:reload(zoxide query -l)+change-preview(ls {})' --bind 'alt-bspace:execute-silent(tmux kill-session -t {})+reload(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/reload_sessions.sh)' --bind bspace:backward-delete-char --bind esc:abort --bind ctrl-n:up --bind ctrl-p:down --bind enter:replace-query+print-query --bind ctrl-u:preview-half-page-up --bind ctrl-d:preview-half-page-down --bind 'ctrl-r:execute(bash -c '\'' printf >&2 "New name: ";read name; tmux rename-session -t {1} "${name}"; '\'')+reload(bash -c '\'' tmux list-sessions | sed -E "s/:.*$//"; '\'')' --bind '?:toggle-preview' --bind change:first --exit-0 '--header=enter=󰿄  alt-bspace=󱂧  ctrl-r=󰑕  ctrl-x=󱃖  ctrl-w=  ctrl-e=󰇘  ctrl-b=󰌍  ctrl-t=󰐆  ctrl-u=  ctrl-d= / ctrl-y=' '--preview=/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {}' --preview-window=top,75%,, --layout=default $'--pointer=\226' -p 75%,85% --prompt ' ' --print-query --tac --scrollbar $'\226\214\226\220' --border-label 'Current session: "" ' --bind 'focus:transform-preview-label:echo [ {} ]' --bind 'false:reload(echo -e "false\ndev")+change-preview(/Users/my-user/.config/tmux/plugins/tmux-sessionx/scripts/preview.sh {1})'
> + RESULT=
> + handle_output ''
> + '[' -d '' ']'
> + is_fzf-marks_mark ''
> ++ echo ''
> ++ grep -q ' : '
> + return 1
> + echo ''
> + grep :
> ++ echo ''
> ++ tr -d '\n'
> + target=
> + [[ -z '' ]]
> + exit 0
> ```

**Comment by rjorgenson** [comment]
Please disregard, I somehow missed https://github.com/omerxx/tmux-sessionx/issues/22#issuecomment-1873494685 and killall and restarting my sessions did the trick for me.

> I don't suppose anyone else has run into this issue? I am on the latest bash in homebrew (`5.2.37(1)-release`) and I am having this exact issue so it doesn't seem to be related to the bash version. I have two running tmux sessions named `1` and `2`. If I have only a single session the script just flashes really quickly and goes away with no output and a `0` return code.
> 
> ```
> #: bash --version
> GNU bash, version 5.2.37(1)-release (aarch64-apple-darwin24.2.0)
> Copyright (C) 2022 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> 
> This is free software; you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> #: bash .tmux/plugins/tmux-sessionx/scripts/sessionx.sh
> .tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
> 2: syntax error in expression (error token is "2")
> .tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
> 2: syntax error in expression (error token is "2")
> .tmux/plugins/tmux-sessionx/scripts/sessionx.sh: line 13: 1
> 2: syntax error in expression (error token is "2")
> #:
> ```

**Comment by rachedhedia** [comment]
I'm still having the same issue (macOS), here are the versions I have installed locally:

![Image](https://github.com/user-attachments/assets/98162088-e9e9-4869-808f-aaa5e70d4ab0)



@omerxx : happy to jump on a call if you want to check this

**Comment by Zananok** [comment]
> Thanks [@heywhy](https://github.com/heywhy). I confirm updating the bash solves the problem. I am closing this issue. However, I suggest adding this to the Readme since the default bash on MacOS was version 3.x.x.

I don't use mac, but this seems the solution.


List of alternatives for anyone in the future:
- Update tmux to latest
- Update Zoxide to latest
- Update bat to latest
- Update bash to latest (regardless if you use zsh, fish, or anything else. The script is using bash)
- Restart tmux (`tmux kill-server`)
- Have nerd-fonts installed and correctly configured for your terminal

- If all fails, try an alternative terminal, as your current terminal could have a slight issue with the scroll.

**Comment by rachedhedia** [comment]
Confirmed, updating bash fixes the issue, thanks

**Comment by koertje** [comment]
tmux 3.3a
fzf 0.62.0 (d226d841)
zoxide v0.4.3-unknown
bat 0.22.1
bash 5.2.15

Nothing happens when I try to use sessionx.
Is there anything I can do or try?
If there's more info you need, just ask.

**Comment by koertje** [comment]
> tmux 3.3a fzf 0.62.0 (d226d841) zoxide v0.4.3-unknown bat 0.22.1 bash 5.2.15
> 
> Nothing happens when I try to use sessionx. Is there anything I can do or try? If there's more info you need, just ask.

I figured out was was going wrong.
Some fzf things don't work in bash if you don't run **`eval "$(fzf --bash)`** first.  So I put that in my bashrc.
Everything works fine... until I open tmux.
Turns out, tmux uses a login shell (and thus doesn't source .bashrc).  So I added the eval thing to .bash_profile and now fzf works in tmux as well.

And now that fzf works in tmux, sessionx works too.

**Comment by rjorgenson-ns** [comment]
> > tmux 3.3a fzf 0.62.0 (d226d841) zoxide v0.4.3-unknown bat 0.22.1 bash 5.2.15
> > 
> > Nothing happens when I try to use sessionx. Is there anything I can do or try? If there's more info you need, just ask.
> 
> I figured out was was going wrong.
> Some fzf things don't work in bash if you don't run **`eval "$(fzf --bash)`** first.  So I put that in my bashrc.
> Everything works fine... until I open tmux.
> Turns out, tmux uses a login shell (and thus doesn't source .bashrc).  So I added the eval thing to .bash_profile and now fzf works in tmux as well.
> 
> And now that fzf works in tmux, sessionx works too.

You can include session things in .bashrc and login things in .bash_profile and just source .bashrc from .bash_profile, then you'll always have all of your settings.
